From f12ae5ab2f0867983ac243ecf6b0f862e72bb190 Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Thu, 13 Jun 2024 09:34:12 +0530
Subject: [PATCH] Fixed build error in A15.

Observed build failure in A15 due to inheritance declaration and
unsupported api.

Tracked-On: NA
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 _studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp | 2 +-
 _studio/shared/include/mfx_utils.h     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp b/_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp
index 3f3876e7..bbc1a4fe 100644
--- a/_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp
+++ b/_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp
@@ -2785,7 +2785,7 @@ mfxStatus  VideoVPPHW::Init(
     {
         CmDevice* pCmDevice = QueryCoreInterface<CmDevice>(m_pCore, MFXICORECM_GUID);
 
-        sts = m_SCD.Init(par->vpp.In.CropW, par->vpp.In.CropH, par->vpp.In.Width, par->vpp.In.PicStruct, pCmDevice, CommonCaps::IsCmSupported(m_pCore->GetHWType()));
+        sts = m_SCD.Init(par->vpp.In.CropW, par->vpp.In.CropH, par->vpp.In.Width, par->vpp.In.PicStruct, pCmDevice);
         MFX_CHECK_STS(sts);
 
         m_SCD.SetGoPSize(ns_asc::Immediate_GoP);
diff --git a/_studio/shared/include/mfx_utils.h b/_studio/shared/include/mfx_utils.h
index bab156a8..e43d7d01 100644
--- a/_studio/shared/include/mfx_utils.h
+++ b/_studio/shared/include/mfx_utils.h
@@ -318,7 +318,7 @@ template <class F>
 struct result_of;
 
 template <typename TRes, typename... TArgs>
-struct result_of<TRes(TArgs...)> : std::result_of<TRes(TArgs...)> {};
+struct result_of<TRes(TArgs...)> {};
 
 template <typename TRes, typename... TArgs>
 struct result_of<TRes(*const&)(TArgs...)>
-- 
2.34.1

